.github/workflows/development-tests.yml

```diff
@@ -0,0 +1,13 @@
+# This workflow runs tests on all non-master branches to ensure code stability and quality before merging.
+
+name: Run Tests
+
+on:
+  push:
+    branches-ignore:
+      - master
+      - release-please--branches--master
```

Comment:
What's the purpose of the `release-please--branches--master` branch?

Reply:
That is the release branch that gets used for release PRs. Line 9 ensures we don't run tests on that branch, since we already run tests in other workflows.

Tests are now run on:
- development branches (any other than `master` and `release-please--branches--master`)
- `master` branch before opening release PR
- `master` branch before publishing

Reply:
Makes sense, thanks for the explanation